include/alpaka/queue/QueueGenericSyclBlocking.hpp

```diff
@@ -0,0 +1,207 @@
+/* Copyright 2021 Jan Stephan
+ *
+ * This file is part of Alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#ifdef ALPAKA_ACC_SYCL_ENABLED
+
+#    include <alpaka/dev/Traits.hpp>
+#    include <alpaka/event/Traits.hpp>
+#    include <alpaka/queue/Traits.hpp>
+#    include <alpaka/queue/sycl/Utility.hpp>
+#    include <alpaka/wait/Traits.hpp>
+
+#    include <CL/sycl.hpp>
+
+#    include <algorithm>
+#    include <iterator>
+#    include <memory>
+#    include <mutex>
+#    include <shared_mutex>
+#    include <vector>
+
+namespace alpaka
+{
+    namespace experimental
+    {
+        template<typename TDev>
+        class EventGenericSycl;
+
+        //! The SYCL blocking queue.
+        template<typename TDev>
+        class QueueGenericSyclBlocking
+        {
+        public:
+            QueueGenericSyclBlocking(TDev const& dev)
+                : m_dev{dev}
+                , m_queue{
+                      dev.m_context, // This is important. In SYCL a device can belong to multiple contexts.
+                      dev.m_device,
+                      {sycl::property::queue::enable_profiling{}, sycl::property::queue::in_order{}}}
+            {
+            }
+
+            auto operator==(QueueGenericSyclBlocking const& rhs) const -> bool
+            {
+                return (m_dev == rhs.m_dev) && (m_event == rhs.m_event);
+            }
```

Comment:
For comparison operators, use hidden friends:
```suggestion
            friend auto operator==(QueueGenericSyclBlocking const& lhs, QueueGenericSyclBlocking const& rhs) const -> bool
            {
                return (lhs.m_dev == rhs.m_dev) && (lhs.m_event == rhs.m_event);
            }
```

Reply:
Should the `operator==` also be `private`?

Reply:
If you define it as hidden friend, access specifiers don't apply :)